### PR TITLE
feat: add "DeleteData" operation to support deleting rows in DataFrame

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2bpm5xq 实现支持行删除的 Operation](https://app.clickup.com/t/2bpm5xq)

**Review by AChenQ — APPROVED** [review]

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -373,6 +373,16 @@ def _set_item_by_slice(self, key: slice, value: "DataFrame") -> None:
...
         for column in self._columns.values():
             column.loc.__delitem__(key)
+        if self.operations is not None:
+            record_key = self._record_key
+            if isinstance(key, int):
+                record_keys = [record_key[key]]  # type: ignore[index]
+            else:
+                record_keys = [
+                    record_key[i]  # type: ignore[index]
+                    for i in range(*key.indices(len(record_key)))  # type: ignore[arg-type]
```
Comment (linjiX) [inline graviti/dataframe/frame.py:383]: `range(len(record_key))[key]`

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -373,6 +373,16 @@ def _set_item_by_slice(self, key: slice, value: "DataFrame") -> None:
     def _del_item_by_location(self, key: Union[int, slice]) -> None:
         for column in self._columns.values():
             column.loc.__delitem__(key)
+        if self.operations is not None:
+            record_key = self._record_key
+            if isinstance(key, int):
+                record_keys = [record_key[key]]  # type: ignore[index]
```
Comment (linjiX) [inline graviti/dataframe/frame.py:379]: where does these ignore comes from?

File: graviti/openapi/data.py [hunk 3]
```diff
@@ -369,6 +369,46 @@ def add_data(
...
+        ...     "graviti-example",
+        ...     "OxfordIIITPet",
+        ...     draft_number = 1,
+        ...     sheet = "train",
+        ...     record_keys = ["123750493121329585", "123750493121329586"],
+        ... )
+
+    """
+    url = f"{url}/v2/datasets/{owner}/{dataset}/drafts/{draft_number}/sheets/{sheet}/data"
+    delete_data_ = {"record_keys": record_keys}
```
Comment (linjiX) [inline graviti/openapi/data.py:407]: do we need a better name for `delete_data_`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 4]
```diff
@@ -373,6 +373,16 @@ def _set_item_by_slice(self, key: slice, value: "DataFrame") -> None:
...
+            else:
+                record_keys = [
```
Comment (linjiX) [inline graviti/dataframe/frame.py:381]: add a comments about doing it by series slicing feature in the future.

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py (hunk 2 above)
Comment (graczhual) [reply graviti/dataframe/frame.py:379]: The record_key might be None

File: graviti/openapi/data.py (hunk 3 above)
Comment (graczhual) [reply graviti/openapi/data.py:407]: `delete_json`

**Review by linjiX — APPROVED** [review]

**Pull request merged**